src/hooks/useAudio.ts

```diff
@@ -0,0 +1,42 @@
+import { SystemStateContext } from '@/store/System/SystemContext';
+import { useContext, useRef, useState } from 'react';
+
+type useAudioProps = {
+  src: string;
+  loop?: boolean;
+};
+
+const useAudio = ({ src, loop = false }: useAudioProps) => {
+  const context = useContext(SystemStateContext);
+
+  if (context === null) {
+    throw new Error('context is null. The useOverlay hook must be inside an OverlayProvider.');
+  }
+
+  const [isPlaying, setIsPlaying] = useState(false);
+
+  const { sound } = context;
+
+  const audioRef = useRef<HTMLAudioElement>(new Audio(src));
+
+  audioRef.current.loop = loop;
+  audioRef.current.volume = sound * 0.01;
+
```

Comment:
`useAudio` 훅의 loop 가 false일 때, **음악 재생이 모두 끝나고 `isPlayingMusic`의 상태가 변하지 않는 이슈**가 있습니다!

아래처럼 ref를 이용해 `ended` 핸들러를 추가하거나, 시멘틱 마크업을 위해 `ControlCenterItem`에 [\<audio>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/audio) 태그를 추가해도 좋을 것 같습니다! 😀
```suggestion
  audioRef.current.loop = loop;
  audioRef.current.volume = sound * 0.01;
  audioRef.current.addEventListener('ended', () => {
    setIsPlaying(false);
  });
```